src/backends/firebird/row-id.cpp

```diff
@@ -10,12 +10,18 @@
 
 using namespace soci;
 
+#ifdef _MSC_VER
+#pragma warning(push)
+#pragma warning(disable:4702) //unreachable code
```

Comment:
FWIW it looks like this warning could be avoided by not having `firebird_rowid_backend` class at all and just throwing directly from `firebird_session_backend::make_rowid_backend()`.